controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -350,6 +373,32 @@ func (r *ROSAControlPlaneReconciler) reconcileDelete(ctx context.Context, rosaSc
 	return ctrl.Result{RequeueAfter: time.Second * 60}, nil
 }
 
+// isMachinePoolsDeleted check if the controlplane related machinePools are deleted.
+func (r *ROSAControlPlaneReconciler) isMachinePoolsDeleted(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope) (bool, error) {
```

Comment:
```suggestion
func (r *ROSAControlPlaneReconciler) ensureMachinePoolsDeleted(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope) (bool, error) {
```

Reply:
I believe isMachinePoolsDeleted  is better naming (commonly function name start with is ) as well as matching the return boolean as an answer.

Reply:
a function starting with `isSomething`  shouldn't be doing any object mutations. Your function does actually delete machinepools not only check if they are deleted or not.

Reply:
okay, I made it deleteMachinePools instead.